pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -314,7 +314,9 @@ def decomposition(wires):
         return decomp_ops
 
     def adjoint(self):
-        return S(wires=self.wires).inv()
+        op = S(wires=self.wires)
+        op.inverse = not self.inverse
+        return op
```

Comment:
minor flyby comment, but I'm curious if the `op.inverse = not self.inverse` can be included in the `qml.adjoint()` code, to avoid needing to duplicate it in each gate?

Reply:
My only concern would be that this logic implicitly implies that the underlying object is Unitary (adjoint = inverse). By implementing it into the `adjoint()` method of the `operation` class. Even though all operations which inherit from this class override this method, it might still be better to leave it out. But if we just want to include it from the perspective of reducing code redundancy, then yes I should be able to move the logic there 👍🏼

Reply:
yep, everything (currently) is unitary :)

Reply:
I suppose my main reason for originally commenting this is less about the abstraction, and more about making the operator class developer friendly.

Currently, `op.inverse` is an implementation detail which is not publicly documented. Meanwhile, `op.adjoint()` is part of the public API that developers should define.

We should try and avoid developers needing to know about implementation details to satisfy the API!

Reply:
@josh146, in moving the logic to the `Operation` class, I also had to update a bunch of the templates which inherit from `Operation` but don't have an `adjoint` method implemented. This could be a good opportunity for coding challenges during the interview process (something like complete the `adjoint` method for the angle embedding template)

Reply:
@josh146, after speaking with @albi3ro and the rest of the Core team we concluded that it would be best to go with the simplest option so I am going to revert to simply overwriting the adjoint method in the non-parametric ops since the current change requires updating the adjoint method for all of the templates

Reply:
no worries! That makes a lot of sense. I believe the templates have already been fixed in `op-refactor` :)